src/pages/filter/D3Plot.svelte

```diff
@@ -0,0 +1,295 @@
+      if (liveDataRep !== undefined && showLive) {
+        recordings.push(liveDataRep);
+        classes.push("live");
+      }
+      color = d3.scaleOrdinal().domain(classes).range(d3.schemeSet3);
+      publicClassList = classes;
+      return {recordings, classes};
+    }
+
+    // Draw chart
+
+      // set the dimensions and margins of the graph
+  const margin = { top: 30, right: 10, bottom: 10, left: 0 }
+  const width = plotSize - margin.left - margin.right;
+  const height = plotSize * 0.75 - margin.top - margin.bottom;
+
+
+    function onInterval(callback: () => void, milliseconds: number) {
+      const interval = setInterval(callback, milliseconds);
+      onDestroy(() => {
+        clearInterval(interval);
+      });
+    }
+
+    onInterval(() => {
+      const {recordings, classes} = createDataRepresentation();
+      drawParallelPlot(recordings, classes, plot);
+    }, 100);
+
+    onMount(() => {
+      notMountedYet = false;
+      // append the svg object for plot
+      plot = d3
+        .select("#parallel-plot-" + filter)
+        .append("svg")
+        .attr("width", width + margin.left + margin.right)
+        .attr("height", height + margin.top + margin.bottom)
+        .append("g")
+        .attr("transform", "translate(" + margin.left + "," + margin.top + ")");
+    });
+
+    // Highlight the specie that is hovered
+    const highlight = function(event: any, gesture: RecordingRepresentation | {gestureClass: string}){
+      const gestureName = gesture.gestureClass
+
+      // first every group turns grey
+      d3.selectAll(".line")
+        .transition().duration(200)
+        .style("stroke", "lightgrey")
+        .style("opacity", "0.2")
+      // Second the hovered specie takes its color
+      d3.selectAll("." + gestureName)
+        .transition().duration(200)
+        .style("stroke", color(gestureName))
+        .style("opacity", "1")
+    }
+
+    // Unhighlight
+    const doNotHighlight = function(){
+      d3.selectAll(".line")
+        .transition().duration(200).delay(1000)
+        .style("stroke", function(gesture: RecordingRepresentation){ return( color(gesture.gestureClass))} )
+        .style("opacity", "1")
+    }
+
+    let plotDrawn = false;
+
+    function drawParallelPlot(data: RecordingRepresentation[], classes: String[], plot: any) {
+      if (notMountedYet) return;
+
+      const opacity = d3.scaleOrdinal().domain(classes as Iterable<string>).range([0.5, 0.5, 0.5, 0.5]);
+
+      const strokeWidth = d3.scaleOrdinal().domain(classes as Iterable<string>).range([4, 4, 4]);
+      // Delete existing plot
+      const dimensions: axis[] = ["x", "y", "z"];
+    
+      // For each dimension, I build a linear scale. I store all in a y object
+      let y: any = {};
+      for (let i in dimensions) {
+        let axis: axis = dimensions[i];
+        y[axis] = d3
+          .scaleLinear()
+          .domain(
+            d3.extent(data, function (gesture: RecordingRepresentation) {
```

Comment:
Here we get a type error because the `domain(...)` function wants an argument of type `Iterable<NumberValue>`, but the function `d3.extent()` possibly returns `[undefined, undefined]`, in which case `undefined` does not overlap with NumberValue. 

We can fix this issue by creating a function which deals with this situation
```ts
    const getExtent = (axis: axis) => {
      const extent = d3.extent(data, function (gesture: unknown) {
        const value: number = (gesture as RecordingRepresentation)[axis];
        return value;
      });
      if (extent[0] === undefined) {
        throw new Error('Unable to find extent of data!');
      }
      return extent;
    };
```
And then we call the getExtent(...) function instead, which handles the situtation where the results is `[undefined, undefined]` by throwing an error
```ts
    for (let i in dimensions) {
      let axis: axis = dimensions[i];
      y[axis] = d3.scaleLinear().domain(getExtent(axis)).range([height, 0]);
    }
```